consensus/polybft/bridge_event_manager.go

```diff
@@ -542,24 +559,47 @@ func (b *bridgeEventManager) multicast(msg interface{}) {
 func (b *bridgeEventManager) GetLogFilters() map[types.Address][]types.Hash {
 	var bridgeMessageResult contractsapi.BridgeMessageResultEvent
 
+	var bridgeMsg contractsapi.BridgeMsgEvent
+
 	return map[types.Address][]types.Hash{
-		b.config.bridgeCfg.ExternalGatewayAddr: {types.Hash(bridgeMessageResult.Sig())},
+		b.config.bridgeCfg.InternalGatewayAddr: {
+			types.Hash(bridgeMsg.Sig()),
+			types.Hash(bridgeMessageResult.Sig())},
 	}
 }
 
 // ProcessLog is the implementation of EventSubscriber interface,
 // used to handle a log defined in GetLogFilters, provided by event provider
 func (b *bridgeEventManager) ProcessLog(header *types.Header, log *ethgo.Log, dbTx *bolt.Tx) error {
-	bridgeMessageResultEvent := &contractsapi.BridgeMessageResultEvent{}
+	var (
+		bridgeMessageResultEvent contractsapi.BridgeMessageResultEvent
+		bridgeMsgEvent           contractsapi.BridgeMsgEvent
+	)
 
-	doesMatch, err := bridgeMessageResultEvent.ParseLog(log)
-	if err != nil {
-		return err
-	}
+	switch log.Topics[0] {
+	case bridgeMessageResultEvent.Sig():
+		doesMatch, err := bridgeMessageResultEvent.ParseLog(log)
+		if err != nil {
+			return err
+		}
 
-	if !doesMatch || b.sourceChainID != bridgeMessageResultEvent.SourceChainID.Uint64() {
-		return nil
-	}
+		if !doesMatch || b.externalChainID != bridgeMessageResultEvent.SourceChainID.Uint64() {
+			return nil
+		}
+
+		return b.state.BridgeMessageStore.removeBridgeEvents(&bridgeMessageResultEvent)
```

Comment:
Shouldn't we remove bridge event only if `Status == true` (meaning that the bridge event is sucessfully executed)?

cc @goran-ethernal

Reply:
Yes. You are right